testsuite/features/support/commonlib.rb

```diff
@@ -751,3 +751,53 @@ def api_unlock
     file.flock(File::LOCK_UN)
   end
 end
+
+# Function to fetch the current list of events
+#
+# @param hostname String The hostname of the system from requested
+def fetch_event_history(hostname)
+  output, _code = get_target('server').run("spacecmd -u admin -p admin system_listeventhistory #{hostname}", check_errors: true)
+  events = []
+  current_event = {}
+
+  output.split("\n").each do |line|
+    case line
+    when /^Id:\s+(\d+)$/
+      current_event[:id] = Regexp.last_match(1).to_i
+    when /^History type:\s+(.+)$/
+      current_event[:history_type] = Regexp.last_match(1)
+    when /^Status:\s+(.+)$/
+      current_event[:status] = Regexp.last_match(1)
+    when /^Completed:\s+(.+)$/
+      begin
+        current_event[:completed] = Time.parse(Regexp.last_match(1))
+      rescue ArgumentError
+        current_event[:completed] = nil
+      end
+    end
+
+    if current_event.key?(:id) && current_event.key?(:completed)
+      events << current_event
+      current_event = {}
+    end
+  end
+
+  events
+end
+
+# Function to get the highest event ID (latest event)
+#
+# @param hostname String The hostname of the system from requested
+def get_last_event_id(hostname)
+  events = fetch_event_history(hostname)
+  last_event = events.max_by { |event| event[:id] }
+  last_event ? last_event[:id] : nil
+end
+
+# Function to trigger the upgrade command
+#
+# @param hostname String The hostname of the system from requested
+# @param package String The package name where it will trigger an upgrade
+def trigger_upgrade(hostname, package)
+  get_target('server').run("spacecmd -u admin -p admin system_upgradepackage #{hostname} #{package} -y", check_errors: true)
```

Comment:
Could you please include upgradePackage method as part of `features/support/namespaces/system.rb` so we use the API instead of spacecmd?

Reply:
I switched everything to spacecmd command because system.schedulePackageInstall or system.schedulePackageUpgrade is not in the documentation and when I tried to use it, it says it doesn't exist

Reply:
When using the other upgrade method:

```
xmlrpc.client.Fault: <Fault -1: 'redstone.xmlrpc.XmlRpcFault: Could not find method: upgradePackage in class: com.redhat.rhn.frontend.xmlrpc.system.SystemHandler with params: [com.redhat.rhn.domain.user.legacy.UserImpl, java.lang.Integer, java.lang.String]'>
```

Reply:
I try to find a way of getting the packageId but don't have a solution. I tried:
 - package_name = client.packages.search.name(session_key, "mgrprxy")
 - client.system.listInstalledPackages(session_key, 1000010004)
 - client.system.listAllInstalledPackages(session_key, 1000010004)
I don't find mgrpxy...

Reply:
Something you can dig is on how system_upgradepackage is developed on spacecmd, as spacecmd use API calls behind it. But ok, no worries, if it becomes hard, using spacecmd should be fine too.